packages/transactional-adapters/transactional-adapter-pg-promise/src/lib/transactional-adapter-pg-promise.ts

```diff
@@ -0,0 +1,35 @@
+import { TransactionalAdapter } from '@nestjs-cls/transactional';
+import { IDatabase, ITask } from 'pg-promise';
+
+export type PgPromiseDbOrTx = IDatabase<unknown> | ITask<unknown>;
+
+export interface PgPromiseTransactionalAdapterOptions {
+    /**
+     * The injection token for the pg-promise instance.
+     */
+    pgPromiseInstanceToken: any;
+}
+
+export class TransactionalAdapterPgPromise
+    implements TransactionalAdapter<PgPromiseDbOrTx, PgPromiseDbOrTx, any>
+{
+    connectionToken: any;
+
+    constructor(options: PgPromiseTransactionalAdapterOptions) {
+        this.connectionToken = options.pgPromiseInstanceToken;
+    }
+
+    optionsFactory = (pgPromiseDbOrTxInstance: PgPromiseDbOrTx) => ({
+        wrapWithTransaction: async (
+            options: any, // FIXME
+            fn: (...args: any[]) => Promise<any>,
+            setClient: (client?: PgPromiseDbOrTx) => void,
+        ) => {
+            return pgPromiseDbOrTxInstance.tx((tx) => {
+                setClient(tx);
```

Comment:
Options should be passed to `.tx` as the first parameter according to https://vitaly-t.github.io/pg-promise/Database.html#tx. Personally, I would only support the object definition here with `mode` and `tag` properties.